packages/@glimmer/runtime/lib/destroyables.ts

```diff
@@ -212,11 +212,15 @@ export function setScheduleDestroyed(fn: (fn: () => void) => void) {
 }
 
 export function isDestroying(destroyable: Destroyable) {
-  return getDestroyableMeta(destroyable).state >= DestroyingState.Destroying;
+  let meta = DESTROYABLE_META.get(destroyable);
```

Comment:
Avoiding `getDestroyableMeta` (which would initialize a "live" meta structure for misses in `DESTROYABLE_META`) here because we intentionally do not want to initialize the `DestroyableMeta` if one did not already exist.